crates/transaction-pool/src/error.rs

```diff
@@ -203,8 +203,15 @@ pub enum InvalidPoolTransactionError {
     #[error("transaction underpriced")]
     Underpriced,
     /// Thrown if the transaction's would require an account to be overdrawn
-    #[error("transaction overdraws from account")]
-    Overdraft,
+    #[error(
+        "transaction overdraws from account, value or fee: {value_or_fee}, balance: {balance}"
+    )]
+    Overdraft {
+        /// Cost transaction is allowed to consume. See `reth_transaction_pool::PoolTransaction`.
+        value_or_fee: U256,
```

Comment:
this isn't quite accurate because this is the cost this tx requires.

imo we can keep overdraft as is, this provides sufficient context

Reply:
this isn't possible if we do the conversion from overdraft error to rpc error